tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+import pandas as pd
+import pandas_flavor as pf
+import pytest
+import numpy as np
+from itertools import chain
+from functools import reduce
+from numpy.lib import recfunctions as rfn
+from string import ascii_lowercase, ascii_uppercase
+from pandas._testing import assert_frame_equal
+from pandas._testing import assert_series_equal
+from typing import Dict
+
+
+@pf.register_dataframe_method
+def expand_grid(df: pd.DataFrame = None,
+                df_key: str = None,
+                others: Dict = None):
+    #check if others is a dictionary
+    if not isinstance(others, dict):
+        #strictly name value pairs
+        #same idea as in R and tidyverse implementation
+        #probably take this out,
+        #as it is covered in expand_grid function
+        raise ValueError("others must be a dictionary")
+    #if there is a dataframe, for the method chaining,
+    #it must have a key, to create a name value pair
+    if df is not None:
+        if not df_key:
+            raise ValueError("dataframe requires a name")
+        else:
+            others.update({df_key: df})
+    dfs, dicts = check_instance(others)
+
+    return grid_computation(dfs, dicts)
+
+
+def check_instance(entry):
```

Comment:
I believe the same comment as above applies here.

Copy/pasting it for immediate visible context: This function should be imported from the library, rather than re-defined here.